Sources/Vehicle/Templates/CarPlayPaginatedListTemplate.swift

```diff
@@ -0,0 +1,98 @@
+import CarPlay
+import Foundation
+import Shared
+
+final class CarPlayPaginatedListTemplate {
+    enum GridPage {
+        case next
+        case previous
+    }
+
+    private var items: [CPListItem]
+    private var currentPage: Int
+    private let title: String
+
+    private let itemsPerPage: Int = CPListTemplate.maximumItemCount
+    private(set) var template: CPListTemplate
+
+    init(title: String, items: [CPListItem]) {
+        self.title = title
+        self.items = items
+        self.currentPage = 0
+        self.template = CPListTemplate(title: title, sections: [])
+    }
+
+    func updateItems(items: [CPListItem], refreshUI: Bool = false) {
+        self.items = items
+        if refreshUI {
+            updateTemplate()
+        }
+    }
+
+    func updateTemplate() {
+        let totalItems = items.count
+        let startIndex = currentPage * itemsPerPage
+        let endIndex = min(startIndex + itemsPerPage, totalItems)
+        let pageItems = Array(items[startIndex ..< endIndex])
+
+        let section = CPListSection(items: pageItems)
+        template.updateSections([section])
+        template.trailingNavigationBarButtons = getPageButtons(
+            endIndex: endIndex,
+            currentPage: currentPage,
+            totalCount: totalItems
+        )
+    }
+
+    private func getPageButtons(endIndex: Int, currentPage: Int, totalCount: Int) -> [CPBarButton] {
+        var barButtons: [CPBarButton] = []
+
+        guard let forwardImage = UIImage(systemName: "arrow.forward"),
+              let backwardImage = UIImage(systemName: "arrow.backward") else { return [] }
+
+        if endIndex < totalCount {
+            barButtons.append(CPBarButton(
+                image: forwardImage,
+                handler: { _ in
+                    self.changePage(to: .next)
+                }
+            ))
+        } else {
+            barButtons
+                .append(CPBarButton(
```

Comment:
does this look better than not including it?

Reply:
In my pinion it does, it prevents the icons to move to a different position when you have just one possible action (forward or backward), if this is not included the user could keep hitting the icon that is in the same position because it thinks it's the same actions while it is not:

https://github.com/home-assistant/iOS/assets/5808343/a8b121cf-55ae-4ff1-a377-678d6d39f208